recipes/binutils/all/conanfile.py

```diff
@@ -126,64 +128,77 @@ def _raise_unsupported_configuration(self, key, value):
 
-    @functools.lru_cache(1)
-    def _configure_autotools(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=self._settings_build.os == "Windows")
+    def generate(self):
         yes_no = lambda tf : "yes" if tf else "no"
-        conf_args = [
-            f"--target={self.options.target_triplet}",
-            f"--enable-multilib={yes_no(self.options.multilib)}",
-            "--with-system-zlib",
-            "--disable-nls",
-            f"--program-prefix={self.options.prefix}",
-            f"exec_prefix={tools.unix_path(self._exec_prefix)}",
-        ]
-        autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return autotools
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--disable-nls")
+        tc.configure_args.append(f"--target={self.options.target_triplet}")
+        tc.configure_args.append(f"--enable-multilib={yes_no(self.options.multilib)}")
+        tc.configure_args.append(f"--with-zlib={self.deps_cpp_info['zlib'].rootpath}")
+        tc.configure_args.append(f"--program-prefix={self.options.prefix}")
+        tc.configure_args.append(f"--exec_prefix=/bin/exec_prefix")
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        autotools = self._configure_autotools()
+        apply_conandata_patches(self)
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
         autotools.install()
 
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"), recursive=True)
+        copy(
+            self,
+            pattern="COPYING*",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+            keep_path=False,
+        )
 
     def package_info(self):
+        target_bindir = os.path.join(self._exec_prefix, str(self.options.target_triplet), "bin")
+        self.cpp_info.bindirs = ["bin", target_bindir]
+
+        absolute_target_bindir = os.path.join(self.package_folder, target_bindir)
+        binaries = os.listdir(absolute_target_bindir)
+        self.output.info(f"Binaries built: {', '.join(binaries)}")
+        for binary_name in binaries:
+            binary = os.path.join(absolute_target_bindir, binary_name)
+            if os.path.isfile(binary):
+                self.output.info(f"Setting {binary_name.upper()}={binary}")
+                self.buildenv_info.define(f"{binary_name.upper()}", binary)
```

Comment:
Thanks for filling me in -- let just leave a trace for future reviewers so it makes sense

```suggestion
                # See https://github.com/conan-io/conan-center-index/pull/14137/files/7ed7a48e2c993bb7e748570ee7ab9d021790c7dc..e93ea8d3d318b0ef8be84816cef615043b5f7fa9#r1022730130 for details
                self.output.info(f"Setting {binary_name.upper()}={binary}")
                self.buildenv_info.define(f"{binary_name.upper()}", binary)
```